presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/execution/thrift/SimpleAddressThriftSqlFunctionExecutionModule.java

```diff
@@ -41,6 +43,8 @@ public SimpleAddressThriftSqlFunctionExecutionModule(Map<Language, SimpleAddress
     public void configure(Binder binder)
     {
         binder.bind(ThriftSqlFunctionExecutor.class).in(SINGLETON);
+        binder.bind(BlockEncodingManager.class).in(SINGLETON);
```

Comment:
question: Is this going to bind a new instance? Should we reuse the instance bound and initialized in the presto-main? Connectors are free to register custom block serializer. We should make sure that for function namespace the same instance, where the custom serializers are bound, is used

Reply:
I think this binds the singleton instance of the class so it should be the same one bound to `FunctionAndTypeManager`. But I don't know guice well. Anyone I should reach out for confirmation?

Reply:
This will be a different object, because the module is eventually installed by classes like `MySqlFunctionNamespaceManagerFactory`, which creates a new Bootstrap object. Singleton only applies to a same injector, instances are different across injectors.

In order to use an existing `BlockEncodingManager` object, we'll need to modify `FunctionNamespaceManagerFactory.create`, so that the object can gets passed in and binded.

Reply:
I'll do a follow up PR to fix this then. Thanks!

Reply:
Fixed in https://github.com/prestodb/presto/pull/15595